src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -585,18 +587,25 @@ bool checkSymbol(Symbol sym, TextSpan memberSpan, SymbolKind kind, out Symbol re
                         return false;
                     }
 
-                    if (sym.Kind == SymbolKind.Method)
+                    if (kind is SymbolKind.Method or SymbolKind.Property)
                     {
                         if (InSpan(sym.GetFirstLocation(), this.syntaxTree, memberSpan))
                         {
                             return true;
                         }
 
-                        // If this is a partial method, the method represents the defining part,
-                        // not the implementation (method.Locations includes both parts). If the
-                        // span is in fact in the implementation, return that method instead.
-                        var implementation = ((MethodSymbol)sym).PartialImplementationPart;
-                        if ((object)implementation != null)
+                        // If this is a partial member, the member represents the defining part,
+                        // not the implementation (member.Locations includes both parts). If the
+                        // span is in fact in the implementation, return that member instead.
+                        if (sym switch
+#pragma warning disable format
```

Comment:
nit: could we link to a tracking issue? Is the problem what's described in [https://github.com/dotnet/roslyn/issues/46464](https://github.com/dotnet/roslyn/issues/46464)?

Reply:
Alternatively, take the `switch` out of the `if`?
```
    Symbol implementation = sym switch
    {
        MethodSymbol method => method.PartialImplementationPart,
        SourcePropertySymbol property => property.PartialImplementationPart,
        _ => throw ExceptionUtilities.UnexpectedValue(sym)
    };

    if (implementation is { } && InSpan(implementation.GetFirstLocation(), this.syntaxTree, memberSpan))
    {
        result = implementation;
        return true;
    }
```

Reply:
I am not sure, I feel like it is more likely to have the same cause as #73251

Reply:
I would be willing to do either including a link or extracting the switch in a follow-up PR. I almost feel a little defiant toward keeping the current factoring since I want the formatter to be fixed :)

Reply:
I'd like it fixed too :-) But all the related issues are currently marked as Backlog :-/ 
It would help to add links and screenshots to get a fix prioritized

Reply:
Done